src/include/planner/create_plan.h

```diff
@@ -33,6 +34,11 @@ class CreatePlan : public AbstractPlan {
   CreatePlan &operator=(const CreatePlan &) = delete;
   CreatePlan(CreatePlan &&) = delete;
   CreatePlan &operator=(CreatePlan &&) = delete;
+  ~CreatePlan() {
+    if (trigger_when) {
+      delete trigger_when;
```

Comment:
When CreateStatement is converted to CreatePlan, this field is copy constructed. So the two `trigger_when` actually point to different positions. But thanks for your sharp mind. Memory is always tricky:)